storage/leia.go

```diff
@@ -136,15 +136,16 @@ func (k *kvBackedLeiaStore) handleRestore(config LeiaBackupConfiguration) error
 					return err
 				}
 			}
-			set = make([]refDoc, 0)
 			return nil
 		})
 	}
 
 	err := collection.Iterate(query, func(ref leia.Reference, value []byte) error {
 		set = append(set, refDoc{ref: ref, doc: value})
 		if len(set) >= limit {
-			return writeDocuments(set)
+			err := writeDocuments(set)
+			set = make([]refDoc, 0)
```

Comment:
should you allocate a slice with cap 100? (micro optimization though)

Reply:
than `len` would already say 100?